logmet/__init__.py

```diff
@@ -150,6 +172,105 @@ def wrap_for_send(messages):
 
         LOG.debug('Metrics sent to logmet successfully')
 
+    def emit_log(self, message):
+        """
+        :param message: string or dict to send to logmet
+        """
+        self._assert_conn()
+
+        if isinstance(message, str):
+            entry = {'message': message}
+        else:
+            entry = dict(message)
+
+        # The tenant ID must be included for the message to be accepted
+        entry['ALCH_TENANT_ID'] = self.space_id
+
+        encoded = self._pack_dict(entry)
+
+        self._send_log(encoded)
+
+    def _pack_dict(self, msg):
+        """
+        :param msg: the dict to pack
+        :return: string in the format
+           '<num_keypars><len_key1><key1><len_val1><val1>...etc'
+        """
+        parts = []
+        total_keys = len(msg)
+        for key, value in msg.iteritems():
+            key = self._encode_unicode(key)
+            value = self._encode_unicode(value)
+            if not value:
+                # Keys without corresponding value can cause problems.
+                total_keys -= 1
+                continue
+            parts.extend([
+                self._pack_int(len(key)),
+                key,
+                self._pack_int(len(value)),
+                value,
+            ])
+
+        return self._pack_int(total_keys) + ''.join(parts)
+
+    def _encode_unicode(self, obj):
```

Comment:
This functions code is in a couple places, and its really more a "to_bytes" method, but I will take care of that after merge.